### ISSUE TITLE
Too slow when saving a model

### ISSUE THREAD

**Issue by helloworldlxb** [fizyr/keras-retinanet#346]
I tried an epoch with resnet-50 on ImageNet VID dataset. However, the saving is quite slow with 25430/512360 in 12 hours. Could you please help me find where the mistake is ?

**Comment by hgaiser** [comment]
The saving of a snapshot? I'm not sure what you mean. Saving of snapshots is handled by Keras / h5py. I'm not sure if this is the correct place for this issue.

**Comment by helloworldlxb** [comment]
@hgaiser Yes, the problem is with the saving of snapshots. It may take weeks to save a snapshot. Maybe there is something wrong with my computer.

**Comment by annw0922** [comment]
I faced the same problem too. Just download it directly using the url by other way, then copy it into the ~/.keras/models

**Comment by hgaiser** [comment]
Are you referring to the saving of a snapshot or downloading of the pretrained imagenet weights? How did you get the value `25430/512360` ? If it's printed in the console, it is probably the downloading of the imagenet weights. You can try to download them manually: https://github.com/fizyr/keras-models/releases

Afterwards, place them in `~/.keras/models` and it should be good to go.

**Comment by helloworldlxb** [comment]
@hgaiser Thank you for your reply. Yes, it is printed in the console. However, it was trained with my own dataset, it should have nothing to do with downloading.

**Comment by de-vri-es** [comment]
By default, the training process starts with pretrained imagenet weights. These weights will be downloaded by the train script if they aren't available.

If following the steps explained by @hgaiser don't work, could you post the output of the train script at the point where something starts taking 12 hours.

**Comment by helloworldlxb** [comment]
@de-vri-es 
This is a screenshot of the console. It has been saving for 3.5 days.
![saving](https://user-images.githubusercontent.com/12216165/37873720-ce8526d0-3054-11e8-917e-d217d426da28.png)

**Comment by hgaiser** [comment]
Oh wow, that's not saving the model, that's running the model on your validation set. Your validation set is huuuuuge apparently (512.360 images). I'd trim it down or disable evaluation all together because this indeed is going to take ages.

**Comment by helloworldlxb** [comment]
@hgaiser Wow, thanks! Now I know the problem.

**Comment by hgaiser** [comment]
Alright I'll consider this resolved then.
